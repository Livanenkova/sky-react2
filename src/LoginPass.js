import  { useState } from "react";
import useInputRequired from "./useInputRequaried"

function LoginPass() {
  const [errorMessageStr, errorMessageSet] = useState('');

  
  const isInputRequired = useInputRequired();

  function setValue(e) {
    const {value} = e.target
    const required = false;
    const result = isInputRequired.onChange(value,required)
    errorMessageSet(result)
  }

  function setError(e) {
    const {value} = e.target
    const required = false;
    const result = isInputRequired.onBlur(value,required)
    errorMessageSet(result)
  }
  
    return (
    <div>
      <label htmlFor='login'>
        Логин:
      <input type="text" onChange={setValue} onBlur={setError} className="login" placeholder='login' id='login' />
      </label>
      <label htmlFor='password'>
        Пароль:
      <input type="password" onChange={setValue} onBlur={setError} className="password" placeholder='password' id='password'/>
      </label>
      <div>{errorMessageStr}</div>
    </div>
    )
}

export default LoginPass