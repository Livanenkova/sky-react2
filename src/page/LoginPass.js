import  { useState,useRef } from "react";
import { Link} from 'react-router-dom'
// import useInputRequired from "../useInputRequaried"

function LoginPass() {
  const [errorMessageStr, errorMessageSet] = useState('');
  const [inputValue, setInputField] = useState({
    login: '',
    password: '',
  })

  const loginRef = useRef(null)
  const passwordRef = useRef(null)

  const inputHandler = (e) => {
    setInputField(() => ({
      ...inputValue,
      [e.target.name]: e.target.value,
    }));
  };

  const submitButton = () => {
    if(!/@/.test(inputValue.login) && !/.com/.test(loginRef.current.value)){
      loginRef.current.focus()
      errorMessageSet(`Логин должен быть примерно таким "@domain.com"`);
      loginRef.current.style.border=" 3px solid red";
    } else if(passwordRef.current.value.length < 5) {

      passwordRef.current.focus()
      passwordRef.current.style.border=" 3px solid red";
      errorMessageSet(`Пароль должен быть длиннее 6 символов!`);
    
    } else {
      errorMessageSet(`Ваш логин: ${inputValue.login}  Ваш пароль: ${inputValue.password}`);
      loginRef.current.style.border="1px solid grey";
      passwordRef.current.style.border="1px solid grey";
    }
  }
  
    return (
    <div className="listView">
      <label htmlFor='login'>
        Логин:
      <input type="email" ref={loginRef} name="login"  onChange={inputHandler}  className="login" placeholder='login' id='login'/>
      </label>
      <label htmlFor='password'>
        Пароль:
      <input type="password" ref={passwordRef} name="password" onChange={inputHandler} className="password" placeholder='password' id='password'/>
      </label>
      <div>{errorMessageStr}</div>
      <button type="button" onClick={submitButton} style={{width:"150px"}}>
            Залогиниться
        </button>
      <Link to="/">Main Page</Link>
    </div>
    )
}

export default LoginPass