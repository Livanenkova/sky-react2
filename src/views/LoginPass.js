import { useState,useRef } from "react";
import { useNavigate } from 'react-router-dom';

function LoginPass(props) {
  console.log('loginPass',props)
  const [errorMessageStr, errorMessageSet] = useState('');
  const navigate = useNavigate();

  const loginRef = useRef(null)
  const passwordRef = useRef(null)

  const inputHandler = (e) => {
    console.log(e.target.value)
    props.setInputField(() => ({
      ...props.inputValue,
      [e.target.name]: e.target.value,
    }));
  };

  const submitButton = () => {
    if(!/@/.test(props.inputValue.login) && !/.com/.test(loginRef.current.value)){
      loginRef.current.focus()
      errorMessageSet(`Логин должен быть примерно таким "@domain.com"`);
      loginRef.current.style.border=" 3px solid red";
    } else if(passwordRef.current.value.length < 5) {

      passwordRef.current.focus()
      passwordRef.current.style.border=" 3px solid red";
      errorMessageSet(`Пароль должен быть длиннее 6 символов!`);
    
    } else {
      errorMessageSet(`Ваш логин: ${props.inputValue.login}  Ваш пароль: ${props.inputValue.password}`);
      loginRef.current.style.border="1px solid grey";
      passwordRef.current.style.border="1px solid grey";
      navigate('/home');
    }
  }

    return (
    
    <div className="listView">
      <label htmlFor='login'>
        Логин:
      <input type="email" ref={loginRef} name="login"  onChange={inputHandler}  className="login" placeholder='login' id='login'/>
      </label>
      <label htmlFor='password'>
        Пароль:
      <input type="password" ref={passwordRef} name="password" onChange={inputHandler} className="password" placeholder='password' id='password'/>
      </label>
      
      <div>{errorMessageStr}</div>

      <button type="button" onClick={submitButton} style={{width:"150px"}}>
            Залогиниться
        </button>
    </div>
    )
}

export default LoginPass