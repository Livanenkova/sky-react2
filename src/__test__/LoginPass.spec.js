/* eslint-disable no-unused-vars */
/* eslint-disable no-undef */
import React from 'react'
import renderer from 'react-test-renderer';
import {Router } from 'react-router-dom';
import LoginPass from '../views/LoginPass';

const { createMemoryHistory } = require("history");


it('отображается корректно', () => {
  const history = createMemoryHistory();
  const tree = renderer.create(<Router location={history.location} navigator={history}><LoginPass/></Router>).toJSON();
  expect(tree).toMatchSnapshot();
});