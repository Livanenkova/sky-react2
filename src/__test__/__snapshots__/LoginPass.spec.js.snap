// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`отображается корректно 1`] = `
<div
  className="listView"
>
  <label
    htmlFor="login"
  >
    Логин:
    <input
      className="login"
      id="login"
      name="login"
      onChange={[Function]}
      placeholder="login"
      type="email"
    />
  </label>
  <label
    htmlFor="password"
  >
    Пароль:
    <input
      className="password"
      id="password"
      name="password"
      onChange={[Function]}
      placeholder="password"
      type="password"
    />
  </label>
  <div />
  <button
    onClick={[Function]}
    style={
      Object {
        "width": "150px",
      }
    }
    type="button"
  >
    Залогиниться
  </button>
</div>
`;
